pkg/services/publicdashboards/service/query.go

```diff
@@ -180,43 +178,6 @@ func (pd *PublicDashboardServiceImpl) buildMetricRequest(dashboard *dashboards.D
 	}, nil
 }
 
-// buildAnonymousUser creates a user with permissions to read from all datasources used in the dashboard
-func buildAnonymousUser(ctx context.Context, dashboard *dashboards.Dashboard, features featuremgmt.FeatureToggles) *user.SignedInUser {
-	datasourceUids := getUniqueDashboardDatasourceUids(dashboard.Data)
```

Comment:
I think you can clean up even more code - `getUniqueDashboardDatasourceUids` seems to only be used here, and that also calls a bunch of functions that are not used elsewhere.